### PR TITLE
Update deploy script to deploy instance only

### PR DESCRIPTION
**Pull request by aodhgan**

### CODE REVIEW

File: .solhintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+contracts/hardhat-dependency-compiler
```
Comment (aodhgan) [inline .solhintignore:1]: Added to get tests to pass solhint error:  
contracts/hardhat-dependency-compiler/@pooltogether/pooltogether-proxy-factory/contracts/GenericProxyFactory.sol
  2:1  error  Compiler version >0.0.0 does not satisfy the ^0.6.0 semver requirement  compiler-version
  3:8  error  Use double quotes for string literals

File: deploy/deploy.ts [hunk 2]
```diff
@@ -92,11 +109,120 @@ const deployFunction: DeployFunction = async function (hre: HardhatRuntimeEnviro
...
+  const aTokenYieldSourceResult = await deploy('ATokenYieldSource', {
     from: deployer,
+    skipIfAlreadyDeployed: true
   });
-  displayResult('ATokenYieldSourceProxyFactory', aTokenYieldSourceProxyFactoryResult);
+  
+  displayResult('ATokenYieldSource', aTokenYieldSourceResult);
+
+  let aTokenYieldSourceContract
+  if(aTokenYieldSourceResult.address){
```
Comment (asselstine) [inline deploy/deploy.ts]: I don't think you need the conditional.  The aTokenYieldSourceResult should always be defined

File: deploy/deploy.ts [hunk 3]
```diff
@@ -92,11 +109,120 @@ const deployFunction: DeployFunction = async function (hre: HardhatRuntimeEnviro
...
+    aTokenYieldSourceContract = await ethers.getContractAt("ATokenYieldSource", aTokenYieldSourceResult.address)
+    dim(`calling mockInitialize()`)
+    //await aTokenYieldSourceContract.mockInitialize()
+    green(`mockInitialize success`)
```
Comment (asselstine) [inline deploy/deploy.ts:126]: ✂️ Looks like dead code / comments

File: deploy/deploy.ts [hunk 4]
```diff
@@ -92,11 +109,120 @@ const deployFunction: DeployFunction = async function (hre: HardhatRuntimeEnviro
...
+  }
+  else{
+    dim(`TestEnvironment! No deployed ATokens. Using Kovan as mock.`)
+    aaveAddressesArray = (JSON.parse(readFileSync("./aave/aaveKovan.json", {encoding: "utf-8"}))).proto
+  }
+
+  // we can filter here for aTokens that we want - by symbol
+  // const aTokenFilter = [aDai]
+
+  if(aTokenYieldSourceContract){ // needed check for typescript
```
Comment (asselstine) [inline deploy/deploy.ts]: I don't think you'll need this check once you eliminate the conditional above

File: deploy/deploy.ts [hunk 5]
```diff
@@ -92,11 +109,120 @@ const deployFunction: DeployFunction = async function (hre: HardhatRuntimeEnviro
...
+    green(`Now deploying aToken proxies`)
+
+    const aTokenYieldSourceInterface = new ethers.utils.Interface((await hre.artifacts.readArtifact("ATokenYieldSource")).abi)
+    let addressesProviderReg: string
+    if(chainId == 1){
+      addressesProviderReg = LENDING_POOL_ADDRESSES_PROVIDER_REGISTRY_ADDRESS_MAINNET
+    }
+    else {
+      addressesProviderReg = LENDING_POOL_ADDRESSES_PROVIDER_REGISTRY_ADDRESS_KOVAN
```
Comment (asselstine) [inline deploy/deploy.ts]: This should be a named account

File: contracts/yield-source/ATokenYieldSource.sol [hunk 6]
```diff
@@ -68,6 +68,12 @@ contract ATokenYieldSource is ERC20Upgradeable, IProtocolYieldSource, AssetManag
   /// @notice Interface for Aave lendingPoolAddressesProviderRegistry
   ILendingPoolAddressesProviderRegistry public lendingPoolAddressesProviderRegistry;
 
+  
+  /// @notice Mock Initializer to prevent 
+  function mockInitialize() public initializer {
```
Comment (asselstine) [inline contracts/yield-source/ATokenYieldSource.sol]: could we rename this to `freeze`?  Mocks are exclusive to testing

File: deploy/deploy.ts [hunk 7]
```diff
@@ -92,11 +109,120 @@ const deployFunction: DeployFunction = async function (hre: HardhatRuntimeEnviro
...
+      green(`aToken proxy for ${aTokenEntry.aTokenSymbol} created at ${createdEvent.args.created}`)
+
+      dim(`saving deployments file for ${aTokenEntry.aTokenSymbol}`)
+
+      let jsonObj: ProxyDeployment = {
+        address: createdEvent.args.created,
+        transactionHash: receipt.transactionHash,
+        receipt: receipt,
+        args: constructorArgs,
+        bytecode: `${prefix}${createdEvent.args.created}${postfix}`
```
Comment (asselstine) [inline deploy/deploy.ts]: Looks like this is just concatenating the '0x' from the address in the middle.

I'd rather see you pull the bytecode directly; i.e. `provider.getCode(address)`

File: hardhat.config.ts [hunk 8]
```diff
@@ -54,6 +59,10 @@ const config: HardhatUserConfig = {
     outDir: 'types',
     target: 'ethers-v5',
   },
+  dependencyCompiler: {
+    paths:['@pooltogether/pooltogether-proxy-factory/contracts/GenericProxyFactory.sol'],
+    keep: true
```
Comment (asselstine) [inline hardhat.config.ts]: I don't think keep is needed...doesn't this just leave artifacts in the contracts/ directory?

**Review by asselstine — COMMENTED** [review]
Looking pretty good.  Just some minor things here and there.

Please add instructions on how to add new aToken markets in the readme, so that when we need to deploy a new one it's easy to do so. How-tos on both:

- Adding an entirely new network
- Adding a new market

File: deploy/deploy.ts (hunk 2 above)
Comment (aodhgan) [reply deploy/deploy.ts]: If i define types on everything then yes!

File: deploy/deploy.ts (hunk 3 above)
Comment (aodhgan) [reply deploy/deploy.ts:126]: This is needed! but instead we should check if the intialization has already happened or not to make this script re-runnable.

Comment (aodhgan) [reply deploy/deploy.ts:126]: Instead checking `newlyDeployed` flag

**Pull request merged**